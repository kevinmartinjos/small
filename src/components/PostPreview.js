import React from 'react';
import {Link} from 'react-router';
import '../styles/PostPreview.css';
import {Panel} from 'react-bootstrap';

var PropTypes = React.PropTypes;

var PostPreview = React.createClass({
	propTypes: {
		title: PropTypes.string.isRequired,
		previewText: PropTypes.string.isRequired,
		id: PropTypes.number.isRequired
	},

	render() {
		return(
			<Link className='title' to={'/posts/' + this.props.id}>
				<Panel className="PostPreview">
					<h3>{this.props.title}</h3>
					<section>{this.props.previewText}</section>
				</Panel>
			</Link>
		);
	}
});

export default PostPreview;
