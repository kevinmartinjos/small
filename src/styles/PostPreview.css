.PostPreview{
	text-align: left;
	margin-left: 20px;
	margin-right: 20px;
}

.title{
	color: rgba(0, 0, 0, 0.8);
	cursor: pointer;
	letter-spacing: -0.02em;
	font-weight: 700;
	font-family: 'medium-content-sans-serif-font,"Lucida Grande","Lucida Sans Unicode","Lucida Sans",Geneva,Arial,sans-serif'
}

.title:hover{
	text-decoration: none;
}

.PostPreview section{
	font-size: 16px;
	line-height: 1.3;
	letter-spacing: -.02em;
	font-weight: 300;
	font-family: 'medium-content-serif-font,Georgia,Cambria,"Times New Roman",Times,serif';
}