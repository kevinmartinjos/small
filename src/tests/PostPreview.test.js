import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import PostPreview from '../components/PostPreview';
import ReactTestUtils from 'react-addons-test-utils'

test('PostPreview should be rendered', () => {
	const component = renderer.create(
		<PostPreview id={1} title="foo" previewText="this is not bar" />
	);
	let tree = component.toJSON();
  	expect(tree).toMatchSnapshot();
});


