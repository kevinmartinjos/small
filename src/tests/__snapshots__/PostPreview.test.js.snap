exports[`test PostPreview should be rendered 1`] = `
<div
  className="PostPreview">
  <h3>
    <a
      onClick={[Function]}
      style={Object {}}>
      foo
    </a>
  </h3>
  <section>
    this is not bar
  </section>
</div>
`;
